### DISCUSSION TITLE
General / custom option like -o customkey=customvalue in typer?

### DISCUSSION THREAD

**Discussion by tddschn** [fastapi/typer#1324] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```
I want my app to do this:

$ my-app -o a=1 -o b=2 -o passwordlogin=false
```

### Description

I'm using my typer app to pass a json body to an API, the body should be formatted like this: 
```
{
  "module1": true
  "module2": false
}
```
where the module names are supplied by the user.

How do I do this in typer?

### Operating System

macOS

### Typer Version

0.4.1

### Python Version

Python 3.10.4

**Comment by Andrew-Sheridan** [comment]
- accept strings with a known pattern eg key<DELIM>value
- split the strings on <DELIM> into tuples
- do stuff with your list of tuples

**Comment by YuriiMotov** [comment, marked as answer]
You can parse options using callback and save parsed key-value items in global object that you can later use

```py
from typing import Annotated, Optional

import typer

options: dict[str, str] = {}

def parse_option(value: list[str]):
    for opt in value:
        k, v = opt.split("=", maxsplit=1)
        options[k] = v
    return value

def main(
    o: Annotated[Optional[list[str]], typer.Option("-o", callback=parse_option)] = None,
):
    typer.echo(options)

if __name__ == "__main__":
    typer.run(main)
```

```
$ python main.py -o a=1 -o b=2 -o passwordlogin=false
{'a': '1', 'b': '2', 'passwordlogin': 'false'}
```
